backend/__tests__/fullSuiteTest.test.js

```diff
@@ -1,70 +1,739 @@
-let server = require('../app').server;
-let socket;
-let address = 'http://localhost:8888/';
 const request = require('supertest');
 const socketIoClient = require('socket.io-client');
-const ioBack = require('socket.io')(server);
-const ioOnConnect = require('../app').ioOnConnect;
+const socketIo = require('socket.io');
+const App = require('../Classes/App').App;
+const Room = require('../Classes/Room').Room;
 
-ioBack.on('connection', (s) => ioOnConnect(s));
+const orgData = require('../testHelper/data.json');
 
-/**
- * Setup WS & HTTP servers
- */
-beforeAll((done) => {
+let spotifyServer = require('../testHelper/spotifyServer').server;
 
-    server.listen(8888, () => {
-        console.log('Server started on port: ' + server.address().port);
-    });
-    
-    socket = socketIoClient(address, {forceNew: true});
+let secTillDelete = 60;
+let env = {
+    backendPort: '8889',
+    frontendPort: '88',
+    ipAddress: 'localhost',
+    spotifyServerAddress: 'localhost',
+    spotifyServerPort: '8890'
+};
+
+
+let spotifyAccountAddress = 'http://' + env.spotifyServerAddress + ':' + env.spotifyServerPort;
+let spotifyApiAddress = spotifyAccountAddress + '/api';
+let Application = new App(
+    false, 
+    env, 
+    secTillDelete, 
+    spotifyAccountAddress, 
+    spotifyApiAddress,
+    500
+);
+
+let ioBack = socketIo(Application.server);
+
+let hostSocket;
+let clientSocket;
+let address = 'http://' + env.ipAddress + ':' + env.backendPort;
+
+ioBack.on('connection', (s) => Application.socketCall(s));
+
+describe('Full Backend Test', () => {
```

Comment:
rename: Server starts properly?

Reply:
This is the container for the main Backend tests, these are the ones that need a server running for them to work, the tests is not only for the server starting